src/main/java/net/skinsrestorer/bungee/utils/SkinApplierBungee.java

```diff
@@ -1,49 +1,62 @@
+/*
+ * #%L
+ * SkinsRestorer
+ * %%
+ * Copyright (C) 2021 SkinsRestorer
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+package net.skinsrestorer.bungee.utils;
 
-import net.md_5.bungee.api.ProxyServer;
 import net.md_5.bungee.api.connection.ProxiedPlayer;
 import net.md_5.bungee.connection.InitialHandler;
 import net.md_5.bungee.connection.LoginResult;
 import net.md_5.bungee.connection.LoginResult.Property;
-import skinsrestorer.shared.storage.Config;
-import skinsrestorer.shared.utils.ReflectionUtil;
+import net.skinsrestorer.bungee.SkinsRestorer;
+import net.skinsrestorer.shared.utils.PlayerWrapper;
+import net.skinsrestorer.shared.utils.ReflectionUtil;
+import net.skinsrestorer.shared.utils.SRLogger;
 
 import java.io.ByteArrayOutputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.util.UUID;
 
-public class SkinApplier {
-    private SkinsRestorer plugin;
-    private static Class<?> LoginResult;
+public class SkinApplierBungee {
+    private final SkinsRestorer plugin;
+    private final SRLogger log;
+    private static Class<?> loginResult;
 
-    public SkinApplier(SkinsRestorer plugin) {
+    public SkinApplierBungee(SkinsRestorer plugin) {
         this.plugin = plugin;
+        this.log = plugin.getSrLogger();
     }
 
     public void applySkin(final ProxiedPlayer p, final String nick, InitialHandler handler) throws Exception {
-        String uuid = null;
         if (p == null && handler == null)
             return;
 
         if (p != null) {
-            uuid = p.getUniqueId().toString();
             handler = (InitialHandler) p.getPendingConnection();
         }
-        Property textures = (Property) plugin.getSkinStorage().getOrCreateSkinForPlayer(nick);
 
-        if (handler.isOnlineMode()) {
-            if (p != null) {
-                sendUpdateRequest(p, textures);
-                return;
-            }
-            // Online mode => get real IP from API
-            uuid = plugin.getMojangAPI().getUUID(nick);
+        Property textures = (Property) plugin.getSkinStorage().getOrCreateSkinForPlayer(nick);
 
-            // Offline mode use offline uuid
-        } else {
-            uuid = UUID.nameUUIDFromBytes(("OfflinePlayer:" + nick).getBytes(StandardCharsets.UTF_8)).toString();
```

Comment:
Think this was a issue on a special version.
I dont know if this version is still used
maybe old parts from 1.7.10 compatibility?